From 03f086531c1b72f95b38a77dc4c59d3211080682 Mon Sep 17 00:00:00 2001
Message-Id: <>
In-Reply-To: <>
References: <>
From: Evgeniy Baskov <>
Date: Mon, 13 Mar 2023 13:43:24 +0300
Subject: [PATCH 28/39] x86/build: set type_of_loader for EFISTUB

After switching to the local copy of bootparams, EFISTUB stopped
setting type_of_loader, using the default value of 0. Restore that
behavior by assigning the right value at the build time.

Signed-off-by: Evgeniy Baskov <>
---
 arch/x86/boot/tools/build.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/arch/x86/boot/tools/build.c b/arch/x86/boot/tools/build.c
index 99cfe7724d4d..cd9fe2bf2c87 100644
--- a/arch/x86/boot/tools/build.c
+++ b/arch/x86/boot/tools/build.c
@@ -586,6 +586,8 @@ int main(int argc, char **argv)
 	memcpy(output + setup_size + efi_boot_params + SETUP_HEADER_OFFSET,
 	       setup_header, 0x290 - SETUP_HEADER_OFFSET
 	       /* == max possible sizeof(struct setup_header) */);
+	/* Set type_of_loader to the one that EFISTUB uses for the local copy */
+	output[setup_size + efi_boot_params + SETUP_HEADER_OFFSET + 0x1E] = 0x21;
 #endif
 
 	/* Calculate and write kernel checksum. */
-- 
2.40.1

